Write down existing excluded categories from twofactorauth.org

Carlgo11 opened the pull request:
I've written down the categories that we've previously decided not to include on twofactorauth.org.

If you think a category shouldn't be excluded then open a new issue or comment in an existing one about the topic and we'll discuss it there. Let's just try and get this merged so that we have a something to reference to in said issues/PRs.

That being said, please let me know if there's any spelling mistakes or possible improvements to the text.
Thanks! 😃

Carlgo11 committed:
Add exclusion categories

Carlgo11 committed:
Small rephrasing

Carlgo11 committed:
Add new lines

Carlgo11 committed:
Reformat and remove section on category 2

Carlgo11 committed:
Remove unnecessary section from category 2

mxxcon reviewed (dismissed):
I completely disagree with the content of this issue and how you are trying to shove it through.
If you want to make such significant content changes to the site, let's do it properly, let's do it atomically.
Each section should be separate PR on which we can discuss merits.
@Carlgo11 I know based on your previous comments to PRs that you have a personal aversion to religious topics, however, that is not the voiced viewpoint of most of the contributors to this project.
I personally think there's absolutely nothing wrong with listing political and religious sites.
There's a difference between religious/political and hate speech sites, and classifying all into one group is WRONG.

dsoegijono commented on EXCLUSION.md:
```diff
@@ -1,3 +1,30 @@
+
+*   #### Pornographic sites
+
+    As the volunteers of [2factorauth](https://github.com/2factorauth) need to go through all sites to validate that the site/service provides 2FA to it's users and verify that the pull request meets our style guidelines we've chosen not   to list pornographic sites.  
```
```suggestion
    As the volunteers of [2factorauth](https://github.com/2factorauth) need to go through all sites to validate that the site/service provides 2FA to it's users and verify that the pull request meets our style guidelines we've chosen not to list pornographic sites.  
```

Removing additional spaces.

dsoegijono commented on EXCLUSION.md:
```diff
@@ -1,3 +1,30 @@
+    As the volunteers of [2factorauth](https://github.com/2factorauth) need to go through all sites to validate that the site/service provides 2FA to it's users and verify that the pull request meets our style guidelines we've chosen not   to list pornographic sites.  
+    We've also chosen to do this so that we don't have to go through potentially disturbing or illegal material.
+
+*   #### Religious and Political sites
```
Let's take out this section and discuss it further. IMO we're here to list whether or not a site as tfa, not to regulate hate speech vs free speech.

> To avoid having to make those decisions we've opted out from listing any Religious or Political sites.

Is this currently an issue? If not, let's make the decision when it becomes one.

Carlgo11 committed:
Apply suggestions from 3632#discussion_r323737976

Carlgo11 committed:
Update EXCLUSION.md

Co-Authored-By: Daphne Soegijono <d.soegijono@gmail.com>

kmpoppe commented on EXCLUSION.md:
```diff
@@ -1,3 +1,25 @@
+
+    -   TwoFactorAuth.org is targeted towards consumers and not website administrators. As of right now, very few general internet users choose to host their own websites and services. Therefore we think that the effort on our part to maintain such a list of self hosted alternatives would outweigh the minimal theoretical change in the internet landscape by listing such services.
+
+    -   If the core project doesn't support two factor authentication but instead through plugins it would mean that we would have to list all plugins that enable two factor authentication for the service. Something that is currently impossible with out website layout.
```
```suggestion
    -   If the core project doesn't support two factor authentication but instead through plugins it would mean that we would have to list all plugins that enable two factor authentication for the service. Something that is currently impossible with our website layout.
```
Typo

Carlgo11 committed:
Fix typo

Co-Authored-By: Kai Michael Poppe <kmpoppe@users.noreply.github.com>

kmpoppe committed:
Update exclusion: Potential controversial sites

As per the discussion in the pull request, this is a long shot on solving the conundrum.

rugk commented on EXCLUSION.md:
```diff
@@ -1,3 +1,31 @@
+*   #### Self-hosted services
+
+    The category "self-hosted services/sites" includes all types of software that's designed to be hosted by the end user (such as [ownCloud](https://owncloud.org/), [NextCloud](https://nextcloud.com/), [Mastodon](https://joinmastodon.org/), [phpBB](https://www.phpbb.com/), [XenForo](https://xenforo.com/) etc).  
+    It also includes [Wordpress.org](https://wordpress.org/) (the open source software) but not [Wordpress.com](https://wordpress.com/) (the company).
```
As discussed in https://github.com/2factorauth/twofactorauth/pull/3550 before, i still disagree with this rule.
Bugzilla is also self-hostable. GitLab too. And you already include them.

Mastodon e.g. has one very very big instance https://mastodon.social/ that has 2FA. (and no, it's no plugin)

kmpoppe replied:
... and #3476 ...

kmpoppe replied:
539f4c47ff7eee2424540e28cf5adf5c621b0b01

rugk commented on EXCLUSION.md:
```diff
@@ -1,3 +1,31 @@
+
+*   #### Self-hosted services
+
+    The category "self-hosted services/sites" includes all types of software that's designed to be hosted by the end user (such as [ownCloud](https://owncloud.org/), [NextCloud](https://nextcloud.com/), [Mastodon](https://joinmastodon.org/), [phpBB](https://www.phpbb.com/), [XenForo](https://xenforo.com/) etc).  
```
Fix casing of "Nextcloud":

```suggestion
    The category "self-hosted services/sites" includes all types of software that's designed to be hosted by the end user (such as [ownCloud](https://owncloud.org/), [Nextcloud](https://nextcloud.com/), [Mastodon](https://joinmastodon.org/), [phpBB](https://www.phpbb.com/), [XenForo](https://xenforo.com/) etc).  
```

Carlgo11 committed:
Fix Nextcloud case

Co-Authored-By: rugk <rugk+git@posteo.de>

kmpoppe committed:
Exclusion to an exclusion

Trying to square the circle of #3476 and #3550 regarding self-hosted sites.

RichJeanes commented on EXCLUSION.md:
```diff
@@ -1,3 +1,41 @@
+
+*   #### Pornographic sites
+
+    As the volunteers of [2factorauth](https://github.com/2factorauth) need to go through all sites to validate that the site/service provides 2FA to it's users and verify that the pull request meets our style guidelines we've chosen not to list pornographic sites.  
```
"Pornographic sites" is ambiguous. Is reddit a ponographic site? It certainly has a lot of pornographic content.
```suggestion
    As the volunteers of [2factorauth](https://github.com/2factorauth) need to go through all sites to validate that the site/service provides 2FA to it's users and verify that the pull request meets our style guidelines we've chosen not to list sites that primarily host and serve pornographic content.  
```

kmpoppe replied:
Resolved with 3024fb2

RichJeanes reviewed (commented):
I think it wouldn't be a bad idea to include a statement somewhere that also makes clear our desire to avoid getting filtered by professional and academic web filters (which is why we don't list gambling sites). Though we do list plenty of gaming services so it's possible we've already made it onto at least some lists, but gambling and other adult sites (eg. porn) would almost certainly get us filtered.

kmpoppe committed:
Clarify intentions of exclusions and what "pornographic" sites are.

As per @RichJeans's comment https://github.com/2factorauth/twofactorauth/pull/3632#pullrequestreview-303531670

kmpoppe committed:
Update EXCLUSION.md

Another update to the self hosted exclusion, clarifying lifting conditions in the first sentence.

RichJeanes commented on EXCLUSION.md:
```diff
@@ -1,3 +1,44 @@
 
+Below is a list of categories and websites that we've opted out from listing on TwoFactorAuth.org.
+
+Apart from the reasons stated below, these exclusions are also intended to prevent TwoFactorAuth.org from being added to any professional or academic web filters, where automatic addition of sites supersedes human checks during which the goal of our project would be weighted higher than the link to a potentially questionable site.
```
```suggestion
One of the primary concerns we seek to address with these exclusions is professional and academic web filters. We believe that the service that this site provides should be as accessible as possible, to help as many people as possible, in as many environments as possible. Dynamic web filters can block websites based solely on the existence of keywords which could lead to this service being filtered for mentioning or linking out to categories of websites that web admins have deemed unacceptable for their domain. While we believe that every site should make an effort to protect their users (which often includes enabling Two Factor Authentication), we also believe that accessibility to this list for all has more value than listing every possible site and service.
```

RichJeanes commented on EXCLUSION.md:
```diff
@@ -1,3 +1,44 @@
+
+*   #### Pornographic sites
+
+    As the volunteers of [2factorauth](https://github.com/2factorauth) need to go through all sites to validate that the site/service provides 2FA to it's users and verify that the pull request meets our style guidelines we've chosen not to list sites that primarily host and serve pornographic content.
```
```suggestion
    As the volunteers of [2factorauth](https://github.com/2factorauth) need to review all submissions to validate that the site/service provides 2FA to its users in addition to verifying that the pull request meets our style guidelines, we've chosen to not list sites that primarily host and serve pornographic content. Given the volunteer driven nature of this project, there is no guarantee that there will always be a collaborator available to review a submission that also does not have an  objection to reviewing such material. Special consideration must also be given due to the potential for content in this category to quickly cross subjective personal boundaries that vary widely from person to person.
```

RichJeanes commented on EXCLUSION.md:
```diff
@@ -1,3 +1,44 @@
+*   #### Pornographic sites
+
+    As the volunteers of [2factorauth](https://github.com/2factorauth) need to go through all sites to validate that the site/service provides 2FA to it's users and verify that the pull request meets our style guidelines we've chosen not to list sites that primarily host and serve pornographic content.
+    We've also chosen to do this so that we don't have to go through potentially disturbing or illegal material.
```
```suggestion
```

RichJeanes commented on EXCLUSION.md:
```diff
@@ -1,3 +1,44 @@
-=================
+# TwoFactorAuth.org excluded categories and websites
 
+Below is a list of categories and websites that we've opted out from listing on TwoFactorAuth.org.
```
```suggestion
Below is a list of categories and websites that we, [the collaborators](https://github.com/orgs/2factorauth/teams/collaborators/members), have opted to not list on TwoFactorAuth.org.
```

mxxcon replied:
collaborators link is 404.
`list` word is listed listed twice.

RichJeanes commented on EXCLUSION.md:
```diff
@@ -1,3 +1,44 @@
+
+    Any site that could damage the reputation of TwoFactorAuth.org and/or lead to any controversy with either the maintainers or users, should not be listed.
+    The group of active maintainers will decide on whether to include or exclude a site within a specific timeframe. Once an exclusion was decided upon, the site will be listed in the section below and - if a pull request exists - it will be closed.
+    This decision will also be in effect for pull requests that already existed before this exclusion documentation was published. 
```
I don't think a "retroactivity" clause is really necessary for this document, especially if it is only going to apply to a single subsection.
```suggestion
```

RichJeanes reviewed (commented):
Mostly some language changes, but I tried to keep the original spirit, if not the letter.

Carlgo11 committed:
Apply suggestions from Rich

Co-Authored-By: Rich Jeanes <RichJeanes@users.noreply.github.com>

kmpoppe reviewed (approved):
Approved after intensive discussion.
Thanks to everyone who has contributed to this.
// Kai
